### DISCUSSION TITLE
Two date parameters in search rather than the one we have now

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#1935] [Ideas]
Currently in the search dropdown list, we have the ability to search for nodes modified AFTER a certain date. This is nice, but I would like to suggest that this be bolstered. I'd like to be able to search for _date created_, _date modified_, and modifiers, _before_ or _after_ these two parameters. If I had to choose just one of these, I'd want _date created_.

If there is currently a way to do this, please let me know.
Thanks
~ Bal

**Comment by macmarrum** [comment]
I'm not sure which search dropdown you are talking about. 

This is what I can see in Freeplnae 1.11.14, which seems to cover "_date created_, _date modified_, and modifiers, _before_ or _after_ these two parameters"

![find_by_date](https://github.com/user-attachments/assets/f4327916-4a8e-4522-bfcf-2b4c1fd41150)

**Reply by wordmuse** [reply]
I didn't notice that before. I wonder if there would be a way to get "created on" and "modified on" modifier. This would provide a higher level of precision that I would find useful.

Thanks much, Mac
~ Bal

**Reply by macmarrum** [reply]
> I wonder if there would be a way to get "created on" and "modified on" modifier.

It's possible via a Script filter (instead of Date filter). The script itself is fairly simple
```groovy
node.lastModifiedAt.format('yyyy-MM-dd') == '2024-07-31'
```
and
```groovy
node.createdAt.format('yyyy-MM-dd') == '2024-07-31'
```

**Reply by wordmuse** [reply]
So I'm not good with coding... but I can follow simple instructions. I am guessing I would copy the text you wrote into a text file and save it as something like, "mod-on.groovy" ? In order to not actually specify a date till I used the script would I just terminate the text with the double equal sign? So...

_node.lastModifiedAt.format('yyyy-MM-dd') ==_

~ Bal

**Reply by macmarrum** [reply]
There's a `Script Filter` drop-down menu option. You need to copy the code and paste in the `Edit script` window.

![2024-08-02_21,18,40](https://github.com/user-attachments/assets/61d8dd13-7fc3-46a8-8e45-35f22f10cebc)

![2024-08-02_21,19,39](https://github.com/user-attachments/assets/e066b2d6-8666-4764-b7c8-338fb2855977)
